src/lib/run-dev-local.js

```diff
@@ -0,0 +1,127 @@
+const cloneDeep = require('lodash.clonedeep')
+const utils = require('./app-helper')
+const dedent = require('dedent-js')
+const rtLib = require('@adobe/aio-lib-runtime')
+const rtLibUtils = rtLib.utils
+const aioLogger = require('@adobe/aio-lib-core-logging')('@adobe/aio-cli-plugin-app:run-dev-local', { provider: 'debug' })
+
+const {
+  OW_CONFIG_RUNTIMES_FILE,
+  OW_JAR_URL, OW_JAR_PATH,
+  OW_LOCAL_APIHOST,
+  OW_LOCAL_NAMESPACE,
+  OW_LOCAL_AUTH,
+  OW_LOCAL_LOG_FILE
+} = require('../lib/owlocal')
+
+const OW_WAIT_INIT_TIME = 2000
+const OW_WAIT_PERIOD_TIME = 500
+const OW_TIMEOUT = 60000
+
+/**
+ * @typedef {object} RunDevLocalObject
+ * @property {string} config the modified dev config
+ * @property {Function} cleanup callback function to cleanup available resources
+ */
+
+/**
+ * Checks the system for pre-requisites to run local Openwhisk, then runs it.
+ *
+ * @param {object} config the app config
+ * @param {Function} [log] function to log application logs
+ * @param {boolean} [verbose=false] set to true to have verbose logging (openwhisk)
+ * @returns {RunDevLocalObject} the RunDevLocalObject
+ */
+async function runDevLocal (config, log = () => undefined, verbose = false) {
+  const devConfig = cloneDeep(config)
+  devConfig.envFile = path.join(config.app.dist, '.env.local')
+  const owJarFile = path.join(config.cli.dataDir, OW_JAR_PATH)
+
+  // take following steps only when we have a backend
+  log('checking if java is installed...')
+  if (!await utils.hasJavaCLI()) {
+    throw new Error('could not find java CLI, please make sure java is installed')
+  }
+
+  log('checking if docker is installed...')
+  if (!await utils.hasDockerCLI()) {
+    throw new Error('could not find docker CLI, please make sure docker is installed')
+  }
+
+  log('checking if docker is running...')
+  if (!await utils.isDockerRunning()) {
+    throw new Error('docker is not running, please make sure to start docker')
+  }
+
+  if (!fs.existsSync(owJarFile)) {
+    log(`downloading OpenWhisk standalone jar from ${OW_JAR_URL} to ${owJarFile}, this might take a while... (to be done only once!)`)
+    await utils.downloadOWJar(OW_JAR_URL, owJarFile)
+  }
+
+  log('starting local OpenWhisk stack...')
+  const owLocalLogFile = OW_LOCAL_LOG_FILE || path.join(config.app.dist, 'openwhisk-local.log.txt')
+  const owExecaOptions = {
+    stdio: [
+      null, // stdin
+      verbose ? fs.openSync(owLocalLogFile, 'w') : null, // stdout
+      'inherit' // stderr
+    ]
+  }
+  const res = await utils.runOpenWhiskJar(owJarFile, OW_CONFIG_RUNTIMES_FILE, OW_LOCAL_APIHOST, OW_WAIT_INIT_TIME, OW_WAIT_PERIOD_TIME, OW_TIMEOUT, owExecaOptions)
+
+  log('setting local openwhisk credentials...')
+  const runtime = {
+    namespace: OW_LOCAL_NAMESPACE,
+    auth: OW_LOCAL_AUTH,
+    apihost: OW_LOCAL_APIHOST
+  }
+  devConfig.ow = { ...devConfig.ow, ...runtime }
+
+  // delete potentially conflicting env vars
+  delete process.env.AIO_RUNTIME_APIHOST
```

Comment:
you're right, these lines can be deleted since for a local run the env file that wskdebug uses is specified by the plugin (in the launch config), and not anything loaded via .env nor the environment. I'll delete those.

Reply:
fixed in 2fdb59f